patch/ros-humble-ur-client-library.patch

```diff
@@ -0,0 +1,1023 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 878ffed..2ef4cf8 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -9,15 +9,16 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+ endif()
+ 
+ option(WITH_ASAN "Compile with address sanitizer support" OFF)
++option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
++set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+ 
+ ##
+ ## Check C++11 support / enable global pedantic and Wall
+ ##
+ include(DefineCXX17CompilerFlag)
+ DEFINE_CXX_17_COMPILER_FLAG(CXX17_FLAG)
+-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic")
+ 
+-add_library(urcl SHARED
++add_library(urcl
+     src/comm/tcp_socket.cpp
+     src/comm/tcp_server.cpp
+     src/control/reverse_interface.cpp
+@@ -51,12 +52,22 @@ add_library(urcl SHARED
+     src/helpers.cpp
+ )
+ add_library(ur_client_library::urcl ALIAS urcl)
+-target_compile_options(urcl PRIVATE -Wall -Wextra -Wno-unused-parameter)
+-target_compile_options(urcl PUBLIC ${CXX17_FLAG})
+-if(WITH_ASAN)
+-  target_compile_options(urcl PUBLIC -fsanitize=address)
+-  target_link_options(urcl PUBLIC -fsanitize=address)
++
++if(MSVC)
++  include_directories(${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/endian)
```

Comment:
I am afraid this does not work as the endian.h is not installed, and it is include in public headers, so any downstream package would fail.

Reply:
See for example the equivalent patch in ros-jazzy https://github.com/RoboStack/ros-jazzy/blob/463363b50b40c681d1176747db387c925113959a/patch/ros-jazzy-ur-client-library.win.patch#L414 .

Reply:
@traversaro Thanks. synced patch with jazzy https://github.com/RoboStack/ros-humble/pull/264/commits/5b2531866c3e03cbaf76fbf5ebc96aa8a260cba0.